Branch for qol enhancements

ffeibl opened the pull request:
fixes #98 , fixes #99

ffeibl committed:
resetBall now delays setting the ball speed by 0.8 seconds

ffeibl committed:
the method set RandomBallSpeed() has been addedd which generates a random angle (between 45 degreees up and down on both sides) for the ball

ffeibl committed:
increased readability and variable designation

cpaech committed:
Reset scores when hiding home menu

Added logic to reset scoreA and scoreB to 0 when the home menu is hidden, ensuring scores are cleared at the start of a new game.

cpaech commented:
Das ist blöd... Es sollte einen min. Angle geben. Sonst kann es passieren, dass der Ball sich einfach von links nach rechts in einer geraden linie bewegt.

cpaech committed:
Center ball on screen after reset

Adjusts ball position in resetBall() to center it based on ball dimensions, ensuring it appears in the middle of the screen after reset.

ffeibl commented:
Klar mache ich, ist mit Ball Anschneiden ja dann eh etwas egal

cpaech committed:
Improve random ball speed calculation logic

Refactored setRandomBallSpeed to generate ball velocity using more precise angle selection and quadrant logic, resulting in more varied initial ball directions. The new implementation calculates both x and y speed components based on the selected arc and radiant.

ffeibl committed:
Starting Angle for setRandomBallSpeed() is between 20 and 55 degrees

ffeibl committed:
Starting Angle for setRandomBallSpeed() is between 20 and 55 degrees

cpaech committed:
Update Controller.java

ffeibl committed:
Starting Angle for setRandomBallSpeed() is between 20 and 55 degrees

ffeibl commented:
wie du das gemacht hat geht das auch, sieht aber komplizierter aus.. wieso nicht einfach den arc gleich richtig generieren?

cpaech commented:
Was hat das jetzt mit dem while auf sich? Erschließt sich mir nicht ganz?

cpaech committed:
Revert "Starting Angle for setRandomBallSpeed() is between 20 and 55 degrees"

This reverts commit 079833a9fc06be31e7ae37b3f8eda81781938a19, reversing
changes made to 71e8f453f76648651773d00ec50c3f581ddffb96.

cpaech committed:
Revert "Starting Angle for setRandomBallSpeed() is between 20 and 55 degrees"

This reverts commit 71e8f453f76648651773d00ec50c3f581ddffb96.

cpaech committed:
Update Controller.java

ffeibl commented:
Achso also ich glaube wir reden gerade sehr aneinander dran vorbei. Du hast ja jetzt zufällig generiert welcher Quadrant genommen wird und dann entsprechend ziehst du von einem Winkel 0 180 360 grad den generierten Winkel ab. Unseren Winkel an sich generieren wir aber schon so das er positiv oder negativ sein kann, heißt das nicht das mit dieser Variation schon immer zwei Variaten der Quadranten abgedeckt werden? Das while war um so lange einen arc zu generieren bis er halt zwischen 20 und 55 grad ist, was, naja unelegant ist aber funktioniert.

ffeibl commented:
Dafür war ja das while.. Jetzt generierst du den arc zwischen 30 und 90 grad was ja bei 90 grad auch nicht schön ist aber auch ist halt jetzt die Abfrage für jeweils oben/unten rechts und oben/unten links doppelt wenn man einen negativen arc erlauben würde-> sin Punkt cos Achsensymmetrisch. So wie jetzt ist das länger aber evtl. besser lesbar deswegen passt schon.
